thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/AnomalyFlattenResource.java

```diff
@@ -23,87 +23,205 @@
-  public static final String ANOMALY_ID = "anomalyId";
+  private final MergedAnomalyResultManager mergedAnomalyResultDAO;
+  private final MetricConfigManager metricConfigDAO;
+  private final DatasetConfigManager datasetConfgDAO;
+
+  private final ExecutorService executor;
+
   public static final String ANOMALY_COMMENT = "comment";
+  private static final String DATAFRAME_VALUE = "value";
+  private static final String DEFAULT_COMMENT_FORMAT = "#%d is %s";
+  private static final int DEFAULT_THREAD_POOLS_SIZE = 5;
+  private static final long DEFAULT_TIME_OUT_IN_MINUTES = 1;
 
-  public AnomalyFlattenResource(MergedAnomalyResultManager mergedAnomalyResultDAO) {
+  public AnomalyFlattenResource(MergedAnomalyResultManager mergedAnomalyResultDAO, DatasetConfigManager datasetConfgDAO,
+      MetricConfigManager metricConfigDAO) {
     this.mergedAnomalyResultDAO = mergedAnomalyResultDAO;
+    this.datasetConfgDAO = datasetConfgDAO;
+    this.metricConfigDAO = metricConfigDAO;
+    this.executor = Executors.newFixedThreadPool(DEFAULT_THREAD_POOLS_SIZE);
   }
 
+
   /**
-   * Returns a list of formatted merged anomalies for UI to render a table
-   * @param detectionConfigId detectionConfigId
+   * Returns a list of formatted metric values and anomaly comments for UI to generate a table
+   * @param metricIdStr a string of metric ids separated by comma
    * @param start start time in epoc milliseconds
    * @param end end time in epoc milliseconds
-   * @param dimensionKeys a list of keys in dimensions; if null, will return all  dimension keys for the anomaly
-   * @return a list of formatted anomalies
+   * @param dimensionStrings a list of keys in dimensions joined by comma
+   * @return a list of formatted metric info and anomaly comments
    */
   @GET
-  @ApiOperation(value = "Returns a list of formatted merged anomalies ")
-  public List<Map<String, Object>> flatAnomalyResults(
-      @ApiParam("detection config id") @QueryParam("detectionConfigId") long detectionConfigId,
+  @ApiOperation(value = "View a collection of metrics and anonalies feedback in a list of maps")
+  @Produces("Application/json")
+  public List<Map<String, Object>> listDimensionValues(
+      @ApiParam("metric config id") @NotNull @QueryParam("metricIds") String metricIdStr,
       @ApiParam("start time for anomalies") @QueryParam("start") long start,
       @ApiParam("end time for anomalies") @QueryParam("end") long end,
-      @ApiParam("dimension keys") @QueryParam("dimensionKeys") List<String> dimensionKeys) {
-    // Retrieve anomalies
-    List<MergedAnomalyResultDTO> anomalies = mergedAnomalyResultDAO.
-        findByStartTimeInRangeAndDetectionConfigId(start, end, detectionConfigId);
+      @ApiParam("dimension keys") @NotNull @QueryParam("dimensionKeys") String dimensionStrings) throws Exception {
+    Preconditions.checkArgument(StringUtils.isNotBlank(metricIdStr));
+    List<String> dimensionKeys = Arrays.asList(dimensionStrings.split(","));
```

Comment:
This is not conformed to the JAX-RS standard. Can pass in a list of `dimensionStrings` and let the JAX-RS handle the null check. It will convert the query parameter `dimensionKey=a&dimensionKey=b` into `java.util.List `automatically.

See more here https://www.mkyong.com/webservices/jax-rs/jax-rs-queryparam-example/

Reply:
Thanks for the comment. It improves my code quality.